mkosi/__init__.py

```diff
@@ -1713,8 +1727,14 @@ def invoke_repart(state: MkosiState, skip: Sequence[str] = [], split: bool = Fal
             definitions.mkdir()
             bootdir = state.root.joinpath("boot/EFI/BOOT")
 
-            if bootdir.exists() and any(bootdir.iterdir()) and any(gen_kernel_images(state)):
-                # If we have at least one kernel images and a bootloader, let's generate an ESP partition.
+            # If Bootable=auto and we have at least one UKI and a bootloader, let's generate an ESP partition.
+            add = (state.config.bootable is True or
+                  (state.config.bootable is None and
+                   bootdir.exists() and
+                   any(bootdir.iterdir()) and
+                   any(gen_kernel_images(state))))
```

Comment:
The `is True` everywhere is a bit unpythonic and I think the code would look better without, but it would leave the `is False` to differentiate that from `None` as the sentinel value for auto. How about adding a tristate enum
```python
class BoolOrAuto(enum.Enum):
     TRUE = True
     FALSE = False
     AUTO = "auto"

     def __bool__(self):
         return bool(self.value)

     def is_auto(self):
         return self.value == "auto"
```
which would make this
```suggestion
            add = (state.config.bootable or 
                   (state.config.bootable.is_auto() and
                    bootdir.exists() and
                    any(bootdir.iterdir()) and
                    any(gen_kernel_images(state))))
```

Reply:
I agree but I'll fix that in a follow up PR